routes/eventsManager.js

```diff
@@ -0,0 +1,167 @@
+const EventModel = require('../models/event');
+
+function getAll(){
+    return new Promise( function (resolve, reject) {
+      try{
+        EventModel.find({},(err,events) => {
+          if(err){
+            reject(err);
+          }else{
+            resolve(events);
+          }
+        });
+      }
+      catch(ex){
+        reject(ex);
+      }
+    });
+  };
+
+function getEventById(_id){
+  return new Promise( function (resolve, reject) {
+    try{
+      EventModel.findById(Number(_id), (err,event) => {
+        if(err){
+          reject(err);
+        }
+        else{
+          resolve(event);
+        }
+      });
+    }
+    catch(ex){
+      reject(ex);
+    }
+  });
+}
+
+function getEventsByTitle(title){
+  return new Promise( function (resolve,reject){
+    try{
+      EventModel.find( { "title": { "$regex": title, "$options": "i" } }, (err,events) =>{
+        if(err){
+          reject(err);
+        }
+        else{
+          resolve(events);
+        }
+      });
+    }
+    catch(ex){
+      reject(ex);
+    }
+  });
+}
+
+function createEvent(_id,title,description,date){
+  return new Promise( function (resolve, reject) {
+    try{
+      EventModel.findById(Number(_id), (err, event) => {
+        if(err){
+          reject(err);
+        }
+        else
+        {
+          if (!event) {
+            const newEvent = EventModel({
+              _id : _id,
+              title: title,
+              description: description,
+              date:  new Date(date)
+            });
+
+            newEvent.save((err) => {
+              if (err) {
+                reject(err);
+              }
+              else
+              {
+                console.log('Inserted 1 event into the collection');
+                resolve(newEvent);
+              }
+            });
+          }
+          else
+          {
+            resolve(undefined);
+          }
+        }
+      });
+    }
+    catch(ex){
+      reject(ex);
+    }
+  });
+}
+
+function updateEvent(_id, title, description, date) {
+  return new Promise( (resolve, reject) => {
+    try {
+      EventModel.findById(Number(_id), (err, event) => {
+        if(err){
```

Comment:
You can clean your code by refactoring conditional statements like this to one line.